artifacts.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/package-registry/archiver"
+)
+
+const artifactsRouterPath = "/epr/{packageName}/{packageName:[a-z_]+}-{packageVersion}.tar.gz"
+
+var errArtifactNotFound = errors.New("artifact not found")
+
+func artifactsHandler(packagesBasePath string, cacheTime time.Duration) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		packageName := vars["packageName"]
+		packageVersion := vars["packageVersion"]
+
+		_, err := semver.Parse(packageVersion)
+		if err != nil {
+			badRequest(w, "invalid package version")
```

Comment:
Would be nice to print out the package version that was invalid here.

Reply:
The version is in URL and I don't expect that the user visits this endpoint anyway. I would leave it as is.